test/test_with_statements.py

```diff
@@ -493,3 +495,35 @@ def test_identifier_syntax():
 
     assert parser.tables == ["hits"]
     assert parser.columns == ["EventDate", "EventTime", "ts_upper_bound"]
+
+
+def test_as_was_preceded_by_with_query():
```

Comment:
@MiuNice, CI complains about the code formatting here. Can you please run `black` here?